auth/api/iam/api.go

```diff
@@ -208,19 +208,52 @@ func (r Wrapper) HandleTokenRequest(ctx context.Context, request HandleTokenRequ
 }
 
 func (r Wrapper) Callback(ctx context.Context, request CallbackRequestObject) (CallbackResponseObject, error) {
-	// check id in path
-	_, err := r.toOwnedDID(ctx, request.Did)
+	// validate request
+	// check did in path
+	ownDID, err := r.toOwnedDID(ctx, request.Did)
 	if err != nil {
-		// this is an OAuthError already, will be rendered as 400 but that's fine (for now) for an illegal id
 		return nil, err
 	}
+	// check if state is present and resolves to a client state
+	if request.Params.State == nil || *request.Params.State == "" {
+		return nil, oauthError(oauth.InvalidRequest, "missing state parameter")
+	}
```

Comment:
If the Callback has an error but is missing the client state, the actual error will not be shown

Reply:
RFC6749 [4.1.2.1](https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.2.1).  Error Response
> state
         REQUIRED if a "state" parameter was present in the client
         authorization request.  The exact value received from the
         client.

I wasn't sure what to do with this, but I will wrap the error if there is one